java/src/main/java/com/powsybl/dataframe/network/NetworkDataframes.java

```diff
@@ -333,8 +333,17 @@ static NetworkDataframeMapper buses(boolean busBreakerView) {
                 .doubles("v_angle", (b, context) -> perUnitAngle(context, b.getAngle()), (b, vAngle, context) -> b.setAngle(unPerUnitAngle(context, vAngle)))
                 .ints("connected_component", ifExistsInt(Bus::getConnectedComponent, Component::getNum))
                 .ints("synchronous_component", ifExistsInt(Bus::getSynchronousComponent, Component::getNum))
-                .strings("voltage_level_id", b -> b.getVoltageLevel().getId())
-                .booleans("fictitious", Identifiable::isFictitious, Identifiable::setFictitious, false)
+                .strings("voltage_level_id", b -> b.getVoltageLevel().getId());
+        if (busBreakerView) {
+            builder.strings("bus_id", b -> {
+                Bus bus = NetworkUtil.getBusViewBus(b);
+                if (bus != null) {
+                    return bus.getId();
+                }
+                return "";
```

Comment:
Maybe I'm too fond of optionals, but that would be rewritten quite nicely if `NetworkUtil.getBusViewBus(b)` was returning an `Optional<Bus>`:
```java
b -> NetworkUtil.getBusViewBus(b).map(Bus::getId).orElse("")
```

Reply:
I agree, fixed